staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery.go

```diff
@@ -477,16 +466,11 @@ func (dm *discoveryManager) RemoveAPIService(apiServiceName string) {
 	}
 }
 
-func (dm *discoveryManager) ExternalServicesSynced() bool {
-	dm.servicesLock.RLock()
-	defer dm.servicesLock.RUnlock()
-	for _, info := range dm.apiServices {
-		if info.lastReconciled.IsZero() {
-			return false
-		}
-	}
-
-	return true
+// Intended Used for testing. Returns true if the queue of services to sync is
+// empty (meaning everything has been reconciled and placed into merged document
+func (dm *discoveryManager) AllServicesSynced() bool {
+	// Once items have successfully synced they are removed from queue.
+	return dm.dirtyAPIServiceQueue.Len() == 0
```

Comment:
This will return true on start up before anything gets added, no? That doesn't seem right.

Reply:
maybe it goes against convention but it works how I'd like. It is meant for testing to check if changes via `AddAPIService`/`RemoveAPIService` have been reconciled yet, so the tests know there are things in the queue before waiting for it to empty.

Ill make it more explicit and move it to tests file to clear that up

Reply:
If it's only for tests, it doesn't matter, sure. If it's to decide if the cached information is a valid summary of the cluster, then it matters a lot (e.g., for healthz or something).